community/procedure-compiler/processor/src/main/java/org/neo4j/tooling/procedure/visitors/ProcedureVisitor.java

```diff
@@ -102,7 +101,11 @@ public ProcedureVisitor( Types typeUtils, Elements elementUtils, boolean ignores
 
     private Stream<CompilationMessage> validatePerformsWriteUsage( ExecutableElement executableElement )
     {
-        return performsWriteVisitor.visit( executableElement );
+        if ( executableElement.getAnnotation( PerformsWrites.class ) != null )
```

Comment:
Do we still support a 'PerformsWrites' annotation, and if so, should it not behave the same as `mode=WRITE`? This method seems specific to only the deprecated version?

Reply:
`PerformsWrites` is deprecated but not removed and can't be paired up with `Procedure.mode` at all. `PerformsWrites` is indeed the same as specifying `mode=WRITE`.
The validation is only required when using the deprecated annotation.